### PR TITLE
Add endpoint for transferring money, bug fix

### PR DESCRIPTION
**Pull request by LimitlessJesse**
Added endpoints for transferring money
Changed schema for accountBalance and amount in both transfer history and expenseHistory to be decimal
Changed how we handle money through sending a record (use math.round for subtracting money for precision)
#34 #35

### CODE REVIEW

**Comment by LimitlessJesse** [comment]
image
Use postman to test the endpoint, route is /user/{senderName}/transfer
Request body as shown in the image

File: back-end/routes/user/index.js [hunk 1]
```diff
@@ -43,4 +43,49 @@ router.get("/:name", async (req, res, next) => {
...
+  try{
+    var errorMessage = "";
+
+    let senderName = req.params['name'];
+    let {receiverName, amount} = req.body;
+    let sender = await User.findOne({ userName: senderName });
+    let receiver = await User.findOne({ userName : receiverName });
+
+    if(receiver != null){
+      if(sender.accountBalance > amount){
```
Comment (afazmunshi49) [inline back-end/routes/user/index.js]: Can we make it "if(sender.accountBalance >= amount)" instead?

Comment (LimitlessJesse) [reply back-end/routes/user/index.js]: Changed it in latest commit

**Review by mike91827 — APPROVED** [review]

**Review by afazmunshi49 — APPROVED** [review]
Looks good

**Pull request merged**